src/blockchain.py

```diff
@@ -461,15 +467,17 @@ def get_next_ips(self, header_hash) -> uint64:
         # 3. Check filter hash is correct TODO
 
         # 4. Check the proof of space hash is valid
-        if (
-            block.header_block.proof_of_space.get_hash()
-            != block.header_block.header.data.proof_of_space_hash
-        ):
-            return False
+        if not pre_verified:
```

Comment:
What if we group the pre validation checks together? Therefore we only have one if statement inside validate_unfinished block and one inside validate_block. This requires reordering the numbers, but I think it's slightly cleaner.

Reply:
yup, that will be cleaner. Also I'll sub-order them from most resource intensive to the least. (return early and avoid expensive validations)